pkg/cluster/internal/providers/podman/provider.go

```diff
@@ -200,3 +202,53 @@ func (p *Provider) node(name string) nodes.Node {
 		name: name,
 	}
 }
+
+// CollectLogs will populate dir with cluster logs and other debug files
+func (p *Provider) CollectLogs(dir string, nodes []nodes.Node) error {
+	execToPathFn := func(cmd exec.Cmd, path string) func() error {
+		return func() error {
+			f, err := common.FileOnHost(path)
+			if err != nil {
+				return err
+			}
+			defer f.Close()
+			return common.ExecToWriter(cmd, f)
+		}
+	}
+	// construct a slice of methods to collect logs
+	fns := []func() error{
+		// TODO(bentheelder): record the kind version here as well
+		// record info about the host podman
+		execToPathFn(
+			exec.Command("podman", "info"),
+			filepath.Join(dir, "podman-info.txt"),
+		),
+	}
+
+	// collect /var/log for each node and plan collecting more logs
+	var errs []error
+	for _, n := range nodes {
+		node := n // https://golang.org/doc/faq#closures_and_goroutines
+		name := node.String()
+		path := filepath.Join(dir, name)
+		if err := internallogs.DumpDir(p.logger, node, "/var/log", path); err != nil {
+			errs = append(errs, err)
+		}
+
+		fns = append(fns, []func() error{
```

Comment:
you can just `append(fns, func(){ ... }, func(){ ... })` instead of creating a slice and then spreading it?

Reply:
:man_facepalming: